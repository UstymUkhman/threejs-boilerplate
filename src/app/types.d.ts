export type AppProps = {
  root: HTMLElement;
};
