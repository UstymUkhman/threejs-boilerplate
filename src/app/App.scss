body > div#stats:last-child {
  left: auto !important;
  right: 0 !important;
}
