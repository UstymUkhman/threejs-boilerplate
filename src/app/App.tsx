import './App.scss';
import Playground from '@/Playground';
import { createSignal } from 'solid-js';
import type { AppProps } from './types.d';
import { Logo, Version } from '@/components';

export default ({ root }: AppProps) => {
  setTimeout(() => setVisibleLogo(false), 2e3);
  const [visibleLogo, setVisibleLogo] = createSignal(true);

  const scene = new Playground().domElement;
  root.appendChild(scene);
  scene.focus();

  return (
    <>
      {visibleLogo() && <Logo />}
      {import.meta.env.DEV && <Version />}
    </>
  );
};
