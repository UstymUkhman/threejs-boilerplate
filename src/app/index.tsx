import './App.scss';
import Sandbox from '@/sandbox';
import type { AppProps } from './types.d';
import { Logo, Version } from '@/components';
import { createSignal, onCleanup } from 'solid-js';

export const App = ({ root }: AppProps) =>
{
  const sandbox = new Sandbox();
  const [logo, hideLogo] = createSignal(true);

  onCleanup(sandbox.dispose.bind(sandbox));
  root.appendChild(sandbox.canvas);

  setTimeout(hideLogo, 2500.0);
  sandbox.canvas.focus();

  return <>
    {logo() && <Logo />}
    {import.meta.env.DEV && <Version />}
  </>;
};
