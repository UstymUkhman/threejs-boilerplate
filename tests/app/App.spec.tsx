import { App } from '@/app';
import { render } from '@solidjs/testing-library';
import { describe, expect, test, vi } from 'vitest';

vi.mock('three/src/renderers/WebGLRenderer');

describe('<App />', () => {
  test('Render', () => {
    const root = document.createElement('main');
    const { container, unmount } = render(()=> <App root={root} />);

    expect(container.innerHTML).toMatchSnapshot();
    unmount();
  });
});
