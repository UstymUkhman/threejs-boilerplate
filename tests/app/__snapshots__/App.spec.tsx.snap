// Vitest Snapshot v1

exports[`<App /> > Render 1`] = `"<div class=\\"_logo_f7d212\\"><img alt=\\"Three.js\\" src=\\"/src/assets/images/logo.png\\"><h1>Three.js Boilerplate</h1></div><aside class=\\"_version_2c80ef\\">v4.0.0</aside>"`;
